pkg/component/controller/workerconfig/reconciler.go

```diff
@@ -469,7 +468,11 @@ func extractAPIServerAddresses(endpoints *corev1.Endpoints) ([]k0snet.HostPort,
 	if len(apiServers) < 1 {
 		// Never update the API servers with an empty list. This cannot
 		// be right in any case, and would never recover.
-		return nil, multierr.Append(errors.New("no API server addresses discovered"), warnings)
+		const msg = "no API server addresses discovered"
+		if warnings := errors.Join(warnings...); warnings != nil {
+			return nil, fmt.Errorf("%s: %w", msg, warnings)
+		}
+		return nil, errors.New(msg)
```

Comment:
Hmm, `warnings := errors.Join(warnings...); warnings != nil` looks confusing. How does it even compile? `warnings` is `[]error` but `errors.Join()` returns `error`, doesn't this make warnings change type mid-sentence? 🤔 


Could the whole block just be `return nil, errors.Join(errors.New("no API server addresses discovered"), warnings...)`?

or

```go
err := errors.New("no API server addresses discovered")
if warnings := errors.Join(warnings...); warnings != nil {
  return nil, fmt.Errorf("%w: %w", err, warnings)
}
return nil, err
```

Reply:
Done.